dataikuapi/dss/ml.py

```diff
@@ -592,32 +757,45 @@ def disable_categories(self, categories, enable_others=False):
 
 class SingleValuedHyperparameterSettings(HyperparameterSettings):
 
-    def __init__(self, name, algo_settings, accepted_values=None, accepted_types=None):
+    def __init__(self, name, algo_settings, accepted_types=None):
         super(SingleValuedHyperparameterSettings, self).__init__(name, algo_settings)
-        self.accepted_values = accepted_values
         self.accepted_types = accepted_types
 
+    def __repr__(self):
+        return self.__class__.__name__ + "(hyperparameter=\"{}\", value={})".format(self.name, self._algo_settings[self.name])
+
+    __str__ = __repr__
+
+    _pretty_repr = __repr__
+
+    def set_value(self, value):
+        if self.accepted_types is not None:
+            assert any(isinstance(value, accepted_type) for accepted_type in self.accepted_types), "Invalid type for hyperparameter {}. Type must be one of: {}".format(self.name, self.accepted_types)
+        self._algo_settings[self.name] = value
+
+
+class SingleValuedCategoricalHyperparameterSettings(HyperparameterSettings):
```

Comment:
The naming suggest that this inherits from `SingleValuedHyperparameterSettings`, can we find a less ambiguous name for `SingleValuedHyperparameterSettings` or maybe factor common behavior in a base class ?

Reply:
There really is not much in common to factor...
We could name them `SingleValueHyperparameterSettings` and `SingleCategoryHyperparameterSettings` ?

Reply:
Looks better !